util/set/set.go

```diff
@@ -4,6 +4,21 @@
 // Package set contains set types.
 package set
 
+// Set is a set of T.
+type Set[T comparable] map[T]struct{}
```

Comment:
make this a struct?

Reply:
naah. I want people to be able to range over it still.

I just want it for better declarations so I know people's intent from the signature/fields